src/compile.rs

```diff
@@ -1574,8 +1546,10 @@ pub fn call_derived_associated_calls<'a>(
                         }) {
                             let call = make_associated_call(resource_name, f);
                             let args = vec![FunctionArgument::new_this_argument(t)];
+                            // TODO: Should there be a parent_context here?  I think this is
+                            // effectively a "fake" context since we're not really parsing the tree
                             let mut local_context =
-                                BlockContext::new(BlockType::Domain, types, Some(t));
+                                BlockContext::new(BlockType::Domain, Some(t), None);
```

Comment:
I am going to ask a question that will hopefully help get to the bottom of this todo.  Can we have chains of associated calls and inheritance?  Something like iptables_tmp inherits tmp_file, tmp_file has some associated calls, tmp_file inherits from any_file, any_file also has some associated calls.  If iptables associates with iptables_tmp, would the inheritance between tmp_file and any_file come into play here?  If it would we may need to include the parent of the thing we are associating with.

Reply:
> Can we have chains of associated calls and inheritance?

Yes.

>  If it would we may need to include the parent of the thing we are associating with.

Do we?  I probably need to think about this scenario more.  Just looking at the immediate context, it looks like all we're setting up is that "this" is a symbol for whatever type is associating here. I guess the real question is whether we'll need to access any bindings (and what scope is even relevant for those bindings?).  My gut says that because we're generating all the call args internally that there can't be any bindings in play here?

I'm not 100% sure though.  Happy to sit down and talk through this case sometime.

Reply:
I guess the proof is in the pudding, if things are working how we want them to this is probably fine as is.  But I am good with bouncing it around if you think we need to.